src/atoms/Checkbox/Checkbox.native.js

```diff
@@ -0,0 +1,153 @@
+import Space from '../Space';
+import spacings from '../../tokens/spacings';
+
+const styles = {
+  backdrop: {
+    width({ width }) {
+      return width;
+    },
+    height({ height }) {
+      return height;
+    },
+    borderRadius({ borderRadius }) {
+      return borderRadius;
+    },
+    backgroundColor({ backgroundColor }) {
+      return backgroundColor;
+    },
+  },
+};
+
+const Backdrop = styled.View`
+  width: ${styles.backdrop.width}px;
+  height: ${styles.backdrop.height}px;
+  border-radius: ${styles.backdrop.borderRadius}px;
+  background-color: ${styles.backdrop.backgroundColor};
+`;
+
+const mapSizeToBackdropProps = (checkBoxSize) => {
+  switch (checkBoxSize) {
+    case 'large':
+      return 24;
+    case 'medium':
+      return 20;
+    case 'small':
+      return 16;
+    case 'xsmall':
+      return 12;
+    default:
+      return 20;
+  }
+};
+
+const mapSizeToHelpTextSizeProp = (checkboxSize) => {
+  switch (checkboxSize) {
+    case 'large':
+      return 'medium';
+    case 'medium':
+      return 'xsmall';
+    default:
+      return 'small';
+  }
+};
+
+const Checkbox = ({ onChange, checked, disabled, size, title, helpText, testID }) => {
+  const [underlayColor, setUnderlayColor] = useState('transparent');
+  const theme = useContext(ThemeContext);
+
+  const onPressIn = () => {
+    const newUnderlayColor = checked ? theme.colors.primary['300'] : theme.colors.tone['400'];
+    setUnderlayColor(newUnderlayColor);
+  };
+
+  const onPressOut = () => {
+    setUnderlayColor('transparent');
+  };
+
+  let titleTextColor = 'shade.700';
+
+  let helpTextColor = 'shade.500';
+
+  if (disabled) {
+    titleTextColor = 'shade.500';
+    helpTextColor = 'shade.300';
+  }
+
+  const backdropSize = mapSizeToBackdropProps(size);
+
+  return (
+    <Flex alignSelf="flex-start">
+      <TouchableOpacity
+        activeOpacity={1}
+        accessibilityRole="checkbox"
+        onPress={onChange}
+        underlayColor="transparent"
+        disabled={disabled}
+        onPressIn={onPressIn}
+        onPressOut={onPressOut}
+        {...automation(testID)}
+      >
+        <Flex flexDirection="row">
+          <View>
+            <Backdrop
+              width={backdropSize}
+              height={backdropSize}
+              borderRadius={backdropSize / 2}
+              backgroundColor={underlayColor}
+            >
+              <CheckBoxIcon checked={checked} size={size} disabled={disabled} />
+            </Backdrop>
+            <Flex alignSelf="center">
+              <Space margin={[0, 0, 0, 0.5]}>
+                <View>
+                  <Text color={titleTextColor} size={size}>
+                    {title}
+                  </Text>
+                </View>
+              </Space>
+            </Flex>
+          </View>
+        </Flex>
+
+        {!isEmpty(helpText) && size !== 'small' && (
+          <Space margin={[0, 0, 0, backdropSize / spacings.unit + 0.5]}>
```

Comment:
you can make it simpler in the structure when you write code.
![IMG_4072](https://user-images.githubusercontent.com/11384858/75227131-a1a6ca80-57d3-11ea-9c25-1ab46c7ecab8.jpg)

So you don't have to mess with spacing everytime

Reply:
I had similar styling earlier, the issue with this is that the Title text and the checkbox icon do not align horizontally, that is because of the size differences of each of the component. 
As per figma file the checkbox and the tile text are center aligned, hence I kept both in the same container with `alignItems: center`.